lib/ansible/galaxy/dependency_resolution/providers.py

```diff
@@ -425,13 +444,15 @@ def is_satisfied_by(self, requirement, candidate):
             requirements=requirement.ver,
         )
 
-    def get_dependencies(self, candidate):
-        # type: (Candidate) -> list[Candidate]
+    def get_dependencies(self, candidate: Candidate) -> t.Generator[Requirement]:
```

Comment:
The `typing.Generator[YieldType, SendType, ReturnType]` generic has 3 arguments, I don't think that having a single value is valid.
However, it's usually unnecessary and using an iterator is a nicer alternative that I've been using everywhere:
```suggestion
    def get_dependencies(self, candidate: Candidate) -> t.Iterator[Requirement]:
```